src/askui_runner/modules/core/infrastructure/files/askui.py

```diff
@@ -1,26 +1,140 @@
-from .s3 import ListObjectsResponse, ListObjectsResponseContent, S3RestApiFilesService
-
-
-class AskUiFilesService(S3RestApiFilesService):
-    HIDDEN_FOLDER_NAME = ".askui"
-
-    def _build_download_file_url(self, remote_file_path: str) -> str:
-        return self._build_file_url(self._remove_workspace_prefix(remote_file_path))
-
-    def _remove_workspace_prefix(self, remote_path: str) -> str:
-        prefix_end = remote_path.find("/", len("workspaces/"))
-        return remote_path[prefix_end + 1 :]
-
-    def _map_list_objects_response(self, content: bytes) -> ListObjectsResponse:
-        response = super()._map_list_objects_response(content)
-        return ListObjectsResponse(
-            continuation_token=response.continuation_token,
-            prefix=response.prefix,
-            contents=[
-                ListObjectsResponseContent(key=content.key)
-                for content in response.contents
-                if not content.key[len(response.prefix or "") :]
-                .lstrip("/")
-                .startswith(self.HIDDEN_FOLDER_NAME)
-            ],
+import os
+import re
+from typing import Optional
+from urllib.parse import urlencode, urljoin
+
+import requests
+from pydantic import BaseModel, Field
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from .files import FilesDownloadService, FilesUploadService
+
+
+class FileDto(BaseModel):
+    name: str
+    path: str
+    url: str
+
+
+class FilesListResponseDto(BaseModel):
+    data: list[FileDto]
+    next_continuation_token: Optional[str] = Field(default=None)
+
+
+REQUEST_TIMEOUT_IN_S=60
+UPLOAD_REQUEST_TIMEOUT_IN_S=3600 # allows for uploading large files
+
+
+class AskUiFilesService(FilesUploadService, FilesDownloadService):
+    HIDDEN_FILES_PATTERNS = [
+        r"^workspaces/[^/]+/test-cases/\.askui/.+$",
```

Comment:
Can I upload files to non `test-cases` folders? 

And do we want to limit the workspace id to an valid UUID?

Reply:
It is not for upload but only download. This is why it is also not necessary to limit to only valid uuids.